factory.py

```diff
@@ -164,4 +163,7 @@ def set_logger(
     def run(self) -> None:
         if self.algorithm is None:
             raise ValueError("No algorithm instance has been created.")
-        self.algorithm.run()
+        if issubclass(type(self.algorithm), VisionAlgorithm):
+            self.algorithm.compress_model()
+        else:
+            self.algorithm.run()
```

Comment:
@Akash-guna 
why don't we inherit all the vision algorithms with the VisionAlgorithm class instead of adding a check here?

Reply:
Don't that already to accomadate this change,

Reply:
Then do we even need this check?

Reply:
Do we have run calling compress model internally in visionalgorithm? If yes then I can remove, will check and remove now

Reply:
hey @shwu-nyunai i am facing issues while trying to revert the last commit, can u just revert it from your end that commit added only this change so it can be safely reverted.

Reply:
@Arnav0400  this is still not done yet. let's wait for the fix before merging.